### ISSUE TITLE
Unable to use exec based values in Kubernetes provider

### ISSUE THREAD

**Issue by CarterCodesRSA** [dflook/terraform-github-actions#57]
Hi @dflook !

This may be a setup error on my part but it seems like the docker image doesn't come with the aws/awscli/aws-iam-authenticator binaries installed, this causes a bit of an issue when using the following setup for a Kubernetes provider:

```
provider "kubernetes" {
  alias = "ops"

  host                   = data.aws_eks_cluster.ops_cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.ops_cluster.certificate_authority[0].data)
  exec {
    api_version = "client.authentication.k8s.io/v1alpha1"
    args        = ["eks", "get-token", "--cluster-name", "ops_cluster"]
    command     = "aws"
  }
}

```

This leads to the terraform plan failing with the following output:
```
Error: Get "https://<Redacted>.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps/aws-auth": getting credentials: exec: executable aws not found
```

**Comment by dflook** [comment]
Hi @CarterCodesRSA

You are right, there aren't any tools installed in the docker image. I haven't used the kubernetes provider with EKS, but is the [eks_cluster_auth](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/eks_cluster_auth) data source an option for getting the token instead of exec?

If you really need to exec inside the container a workaround could be have exec run a script which installs the tools if required and then runs the get-token command. The docker image is based on debian:buster-slim and is invoked as root.

**Comment by CarterCodesRSA** [comment]
Hi!

You're an absolute champ for such fast responses!

So the issue that crops up is when using that data source to provide the auth token, is that 9 times out of 10 the token stored in the terraform state is expired :(

Would it be possibly to have the aws tools installed on the base image? As I'm sure that there are other members of the community that could benefit from having it.

Else, if that is not possible, the issue becomes how to install additional tools on the container as the container is started within the "terraform plan" job?

**Comment by dflook** [comment]
I'm reluctant to install tools in the image, as it will never be the right tools for everyone.

My idea for installing with a script is to create an `aws` wrapper script alongside your terraform:
```bash
#!/bin/bash

(if ! command -v aws >/dev/null; then
    echo "Installing awscli..."

    curl -s "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" >/dev/null
    unzip awscliv2.zip >/dev/null
    ./aws/install
fi) </dev/null

exec aws "$@"
```

And then change your `exec` command to use it:
```hcl
provider "kubernetes" {
  alias = "ops"

  host                   = data.aws_eks_cluster.ops_cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.ops_cluster.certificate_authority[0].data)
  exec {
    api_version = "client.authentication.k8s.io/v1alpha1"
    args        = ["eks", "get-token", "--cluster-name", "ops_cluster"]
    command     = "${path.module}/aws"
  }
}
```

But this is completely untested!

**Comment by CarterCodesRSA** [comment]
I completely understand! 

I'll give it a go and see what I see if that works, if so I will update the issue with if it works.

Thank you again for the help! :D

**Comment by alec-pinson** [comment]
Hi, 
Firstly thank you so much for this action @dflook! 

Sorry for commenting on a closed issue but it kinda relates for us also.. 
At the moment we have this in our terraform docker image

```docker
# Install latest Azure CLI
RUN curl -skL https://aka.ms/InstallAzureCLIDeb | sudo bash

# Install postgres client
RUN apt-get install -y --no-install-recommends postgresql-client
```

I'm not sure if this is possible but if so it would be great if we could maybe do something like:-
```yaml
      - name: terraform plan
        uses: dflook/terraform-plan@v1
        with:
          path: my-terraform-config
          run: |
            # Install latest Azure CLI
            curl -skL https://aka.ms/InstallAzureCLIDeb | sudo bash
            
            # Install postgres client
            apt-get install -y --no-install-recommends postgresql-client
```

I guess the best solution (speed wise) would be if we could somehow supply our own docker image but no idea how that would be possible at all

**Comment by alec-pinson** [comment]
FYI I've created a PR https://github.com/dflook/terraform-github-actions/pull/68 it's untested but will try to test asap

**Comment by dflook** [comment]
The `TERRAFORM_PRE_RUN` environment variable was released in v1.13.0, which makes installing required tools a lot easier
